chia/data_layer/data_layer_wallet.py

```diff
@@ -1138,9 +1137,11 @@ async def finish_graftroot_solutions(offer: Offer, solver: Solver) -> Offer:
                             ]
                         )
                     )
-                    new_spend: CoinSpend = spend.replace(solution=new_solution.to_serialized())
-                    spend = new_spend
-            new_spends.append(spend)
+                    new_spends.append(spend.replace(solution=new_solution.to_serialized()))
+                else:
+                    new_spends.append(spend)
+            else:
+                new_spends.append(spend)
```

Comment:
this feels like just enough awkward to be worth a new variable to avoid it.  to better express "we are always adding something".  maybe `spend_to_add = spend` at the top and


```suggestion
                    spend_to_add = spend.replace(solution=new_solution.to_serialized())
            new_spends.append(spend_to_add)
```

Reply:
Taken